uaclient/tests/test_apt.py

```diff
@@ -101,46 +101,63 @@ def test_valid_apt_credentials_true_when_missing_apt_helper(
     @mock.patch('uaclient.apt.os.unlink', return_value=True)
     @mock.patch('uaclient.util.subp')
     @mock.patch('uaclient.apt.os.path.exists', return_value=True)
-    def test_valid_apt_credentials_returns_true_on_valid_creds(
+    def test_passes_on_valid_creds(
             self, m_exists, m_subp, m_unlink):
-        """Return true when apt-helper succeeds in authentication to repo."""
+        """Succeed when apt-helper succeeds in authenticating to repo."""
 
         # Success apt-helper response
         m_subp.return_value = 'Get:1 https://fakerepo\nFetched 285 B in 1s', ''
 
-        assert True is valid_apt_credentials(
+        assert None is assert_valid_apt_credentials(
             repo_url='http://fakerepo', username='user', password='pwd')
         exists_calls = [mock.call('/usr/lib/apt/apt-helper'),
                         mock.call('/tmp/uaclient-apt-test')]
         assert exists_calls == m_exists.call_args_list
         apt_helper_call = mock.call(
             ['/usr/lib/apt/apt-helper', 'download-file',
              'http://user:pwd@fakerepo/ubuntu/pool/',
-             '/tmp/uaclient-apt-test'], capture=False)
+             '/tmp/uaclient-apt-test'])
         assert [apt_helper_call] == m_subp.call_args_list
         assert [mock.call('/tmp/uaclient-apt-test')] == m_unlink.call_args_list
 
+    @pytest.mark.parametrize(
+        'exit_code,stderr,error_msg',
+        ((1, 'something broke',
+          "Failed running command 'apt-helper ' [exit(1)]. "
+          "Message: something broke"),
+         (100, 'E: Failed to fetch ... HttpError401 on trusty',
+          'Invalid APT credentials provided for http://fakerepo'),
+         (100, 'E: Failed to fetch ... 401 Unauthorized on xenial',
+          'Invalid APT credentials provided for http://fakerepo'),
+         (100, 'E: Failed to fetch ... Connection timed out',
+          'Timeout trying to access http://fakerepo')))
```

Comment:
This is quite a generic error message. I wonder if this message might be clearer as: 'Timeout trying to access APT repository at http://fakerepo'